File: Algorithm/Programmers/Python/Level 2/타겟 넘버.py
def solution(numbers, target):
    global answer
    global number
    number = numbers.copy()
    
    answer = 0
    recursive(len(numbers), 0, target)
    
    return answer

def recursive(idx, record, target):
    if idx == 0:
        global answer
        
        if record == target:
            answer += 1
            
    else:
        global number
        recursive(idx-1, record + number[idx - 1], target)
        recursive(idx-1, record - number[idx - 1], target)

File: Algorithm/Programmers/Java/Level 2/타겟_넘버.java
class 타겟_넘버 {
    int limit;
    int[] numbers;
    int target;
    int answer = 0;
    
    public int solution(int[] numbers, int target) {
        this.limit = numbers.length;
        this.numbers = numbers;
        this.target = target;
        
        dfs(0, 0);
        return answer;
    }
    
    void dfs(int sum, int depth) {
        if(depth == limit) {
            if(sum == target) {
                answer++;
            }
            return;
        }
        
        dfs(sum + numbers[depth], depth + 1);
        dfs(sum - numbers[depth], depth + 1);
    }
}